cypress/integration/4_findings.spec.js

```diff
@@ -0,0 +1,170 @@
+    );
+    cy.contains('test detector');
+    cy.contains('Active');
+
+    // Visit Findings page
+    cy.visit(`${Cypress.env('opensearch_dashboards')}/app/${PLUGIN_NAME}#/findings`);
+
+    // need to wait here specifically to ensure findings are generated
+    cy.wait(10000);
+
+    // Confirm arrival at Findings page
+    cy.url().should(
+      'include',
+      'http://localhost:5601/app/opensearch_security_analytics_dashboards#/findings'
+    );
+
+    // Ingest a new document
+    cy.ingestDocument('cypress-test-windows', sample_document);
+
+    // wait for detector interval to pass
+    cy.wait(60000);
+
+    // Click refresh
+    cy.get('button').contains('Refresh').click({ force: true });
+
+    // Check for non-empty findings list
+    cy.contains('No items found').should('not.exist');
+
+    // Check for expected findings
+    cy.contains('test detector');
+    cy.contains('Windows');
+    cy.contains('Low');
+  });
+
+  it('displays finding details flyout when user clicks on Finding ID or View details icon', () => {
+    // Click findingId to trigger Finding details flyout
+    cy.get(`[data-test-subj="findings-table-finding-id"]`).click();
+
+    // Confirm flyout contents
+    cy.contains('Finding details');
+    cy.contains('Rule details');
+    cy.contains('Search Findings').should('not.exist');
+
+    // Close Flyout
+    cy.get(`[data-test-subj="close-finding-details-flyout"]`).then(($el) => {
+      cy.get($el).click({ force: true });
+    });
+
+    // wait for toasts to clear - in this case, timeout will not work.  Icon cannot be found behind error toasts.
+    cy.wait(10000);
+
+    // Click View details icon
+    cy.get(`[data-test-subj="view-details-icon"]`).click({ force: true });
+
+    // Confirm flyout contents
+    cy.contains('Finding details');
+    cy.contains('Rule details');
+    cy.contains('Search Findings').should('not.exist');
+
+    // Close Flyout
+    cy.get(`[data-test-subj="close-finding-details-flyout"]`).then(($el) => {
+      cy.get($el).click({ force: true });
+    });
+  });
+
+  it('allows user to view details about rules that were triggered', () => {
+    // open Finding details flyout via finding id link
+    cy.wait(5000);
+    cy.get(`[data-test-subj="view-details-icon"]`).click({ force: true });
+
+    // Second rule details - open
+    cy.get('.euiAccordion__button')
+      .contains('Setting Change in Windows Firewall with Advanced Security')
+      .click({ force: true });
+
+    // Confirm content
+    cy.contains('Setting have been change in Windows Firewall');
+    cy.contains('Low');
+    cy.contains('Windows');
+    cy.contains('cypress-test-windows');
+
+    ruleTags.forEach((tag) => {
+      cy.contains(tag);
+    });
+
+    // Close Flyout
+    cy.get(`[data-test-subj="close-finding-details-flyout"]`).then(($el) => {
+      cy.get($el).click({ force: true });
+    });
+
+    // need to wait for error toasts to dissipate, neither icon nor finding id are clickable without this
```

Comment:
Are these errors expected?

Reply:
Currently yes, until error handling has been refactored.  From a user perspective no, they are not expected.